### PR TITLE
ledger: get app list

### PR DESCRIPTION
**Pull request by dviramontes**

### CODE REVIEW

File: packages/hdwallet-ledger-webusb/src/transport.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import Btc from '@ledgerhq/hw-app-btc'
 import getAppAndVersion from '@ledgerhq/live-common/lib/hw/getAppAndVersion'
 import getDeviceInfo from '@ledgerhq/live-common/lib/hw/getDeviceInfo'
 import openApp from '@ledgerhq/live-common/lib/hw/openApp'
+import { listApps } from "@ledgerhq/live-common/lib/apps/hw"
```
Comment (kaladinlight) [inline packages/hdwallet-ledger-webusb/src/transport.ts]: What is the difference between this `apps/hw` implementation and the `/hw` implementation here: https://github.com/LedgerHQ/ledger-live-common/blob/master/src/hw/listApps.js

Comment (dviramontes) [reply packages/hdwallet-ledger-webusb/src/transport.ts]: Good questions. @ryankeepkey mentioned this. The one at  https://github.com/LedgerHQ/ledger-live-common/blob/master/src/hw/listApps.js uses the native transport vs the one we are using we pass it our transport. Not sure what the difference is beyond that but something to keep in mind the one we are using here doesn't work.

File: packages/hdwallet-ledger/src/ledger.ts [hunk 2]
```diff
@@ -353,6 +354,13 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
     }
   }
 
+  public async listApps (): Promise<any> {
+    const deviceInfo = await getDeviceInfo(this.transport)
+    const res = await this.transport.call(null, 'listApps', deviceInfo)
+    console.log({ res })
```
Comment (keepkeyjon) [inline packages/hdwallet-ledger/src/ledger.ts]:
```suggestion
```

File: packages/hdwallet-ledger/src/ledger.ts [hunk 3]
```diff
@@ -353,6 +354,13 @@ export class LedgerHDWallet implements core.HDWallet, core.BTCWallet, core.ETHWa
     }
   }
 
+  public async listApps (): Promise<any> {
```
Comment (keepkeyjon) [inline packages/hdwallet-ledger/src/ledger.ts]: Is it possible to be more specific with the return type? i.e. `Promise<string[]>` or something?

File: packages/hdwallet-ledger-webusb/src/transport.ts (hunk 1 above)
Comment (kaladinlight) [reply packages/hdwallet-ledger-webusb/src/transport.ts]: I am pretty sure both of them pass the transport. That is the way we set up those calls. We wouldn't need to do the `getDeviceInfo` with the `/hw` either. The rest of the utility functions we are using are also in `/hw`. I was surprised to even see the other implementation of `listApps` in `apps/hw`.

**Comment by BitHighlander** [comment]
stale

**Pull request closed without merging**